src/main/java/rapgod/utils/Parser.java

```diff
@@ -0,0 +1,92 @@
+package rapgod.utils;
+
+import rapgod.tasks.Task;
+import rapgod.tasks.ToDo;
+import rapgod.tasks.Deadline;
+import rapgod.tasks.Event;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+
+    /**
+     * Parses the input string to a LocalDateTime if the input contains both date and time.
+     * If the input contains only a date, a default time of 0000 is appended and parsed.
+     * Supports multiple date and time formats.
+     *
+     * @param input The input string to be parsed.
+     * @return A LocalDateTime object representing the parsed date and time.
+     * @throws IllegalArgumentException If the input format is invalid.
+     */
+    public static LocalDateTime parseToDateTime(String input) {
+        DateTimeFormatter dateTimeFormatter1 = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+        DateTimeFormatter dateTimeFormatter2 = DateTimeFormatter.ofPattern("MMM dd yyyy");
+        DateTimeFormatter dateTimeFormatter3 = DateTimeFormatter.ofPattern("MMM dd yyyy h:mma");
+        DateTimeFormatter dateTimeFormatter4 = DateTimeFormatter.ofPattern("MMM dd yyyy hh:mma");
+
+        try {
+            return LocalDateTime.parse(input, dateTimeFormatter1);
+        } catch (DateTimeParseException e1) {
+            try {
+                return LocalDateTime.parse(input + " 0000", dateTimeFormatter1);
+            } catch (DateTimeParseException e2) {
+                try {
+                    LocalDate date = LocalDate.parse(input, dateTimeFormatter2);
+                    return date.atStartOfDay();
+                } catch (DateTimeParseException e3) {
+                    try {
+                        return LocalDateTime.parse(input, dateTimeFormatter3);
+                    } catch (DateTimeParseException e4) {
+                        try {
+                            return LocalDateTime.parse(input, dateTimeFormatter4);
+                        } catch (DateTimeParseException e5) {
+                            throw new IllegalArgumentException("Invalid date or time format. Please use formats like 'dd/MM/yyyy HHmm' or 'MMM dd yyyy'.");
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
There has got to be a better way of formatting this.

Reply:
Since each block tries different date formats, you can make them into separate helper methods eg parseWithFormat1(String input)